dallinger/recruiters.py

```diff
@@ -81,6 +82,11 @@ def reward_bonus(self, assignment_id, amount, reason):
         """Throw an error."""
         raise NotImplementedError
 
+    def finalize_hit(self, assignment_id):
```

Comment:
@MatthewWilkes and I discussed the flexibility we might gain from moving to a more notification-based rather than imperative API style for the recruiter: telling the recruiter what's happened, rather than telling it what to do. This might allow us to reduce the surface area of the API and do less guessing about the specific actions recruiters need to take. There's currently a mix of these styles, and it's unclear where the boundaries are between them
```
award_bonus() # imperative
approve_hit() # imperative
finalize_hit() # actually a notification, but named imperatively
notify_completed() # notification
```